internal/apis/kfd/v1alpha2/distribution/rules/builder.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2017-present SIGHUP s.r.l All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package rules
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	yamlx "github.com/sighupio/furyctl/internal/x/yaml"
+)
+
+var ErrReadingRulesFile = errors.New("error while reading rules file")
+
+type DistroRulesSpec struct {
+	Distribution []Rule `yaml:"distribution"`
+}
+
+type Rule struct {
+	Path      string `yaml:"path"`
+	Immutable bool   `yaml:"immutable"`
+}
+
+type Builder interface {
```

Comment:
this is looks a bit more like a factory than a builder, but I would probably call it Extractor or ImmutablesExtractor?